web/wp-content/themes/dctx-v2/inc/template-tags.php

```diff
@@ -0,0 +1,540 @@
+		$aria_labelledby = ' aria-labelledby="' . $block_title_id . ' ' . $desc_id . '"';
+		$aria_hidden     = '';
+	}
+
+	// Set SVG parameters.
+	$fill   = ( $args['fill'] ) ? ' fill="' . $args['fill'] . '"' : '';
+	$height = ( $args['height'] ) ? ' height="' . $args['height'] . '"' : '';
+	$width  = ( $args['width'] ) ? ' width="' . $args['width'] . '"' : '';
+
+	// Start a buffer...
+	ob_start();
+	?>
+
+	<svg
+	<?php
+		echo dctx_get_the_content( $height ); // WPCS XSS OK.
+		echo dctx_get_the_content( $width ); // WPCS XSS OK.
+		echo dctx_get_the_content( $fill ); // WPCS XSS OK.
+	?>
+		class="icon icon-<?php echo esc_attr( $args['icon'] ); ?>"
+	<?php
+		echo dctx_get_the_content( $aria_hidden ); // WPCS XSS OK.
+		echo dctx_get_the_content( $aria_labelledby ); // WPCS XSS OK.
+	?>
+		role="img">
+		<title id="<?php echo esc_attr( $block_title_id ); ?>">
+			<?php echo esc_html( $block_title ); ?>
+		</title>
+
+		<?php
+		// Display description if available.
+		if ( $args['desc'] ) :
+		?>
+			<desc id="<?php echo esc_attr( $desc_id ); ?>">
+				<?php echo esc_html( $args['desc'] ); ?>
+			</desc>
+		<?php endif; ?>
+
+		<?php
+		// Use absolute path in the Customizer so that icons show up in there.
+		if ( is_customize_preview() ) :
+		?>
+			<use xlink:href="<?php echo esc_url( get_parent_theme_file_uri( '/assets/images/svg-icons.svg#icon-' . esc_html( $args['icon'] ) ) ); ?>"></use>
+		<?php else : ?>
+			<use xlink:href="#icon-<?php echo esc_html( $args['icon'] ); ?>"></use>
+		<?php endif; ?>
+
+	</svg>
+
+	<?php
+	// Get the buffer and return.
+	return ob_get_clean();
+}
+
+/**
+ * Trim the title length.
+ *
+ * @param array $args Parameters include length and more.
+ *
+ * @author WDS
+ * @return string
+ */
+function dctx_get_the_title( $args = array() ) {
+
+	// Set defaults.
+	$defaults = array(
+		'length' => 12,
+		'more'   => '...',
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Trim the title.
+	return wp_trim_words( get_the_title( get_the_ID() ), $args['length'], $args['more'] );
+}
+
+/**
+ * Limit the excerpt length.
+ *
+ * @param array $args Parameters include length and more.
+ *
+ * @author WDS
+ * @return string
+ */
+function dctx_get_the_excerpt( $args = array() ) {
+
+	// Set defaults.
+	$defaults = array(
+		'length' => 20,
+		'more'   => '...',
+		'post'   => '',
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Trim the excerpt.
+	return wp_trim_words( get_the_excerpt( $args['post'] ), absint( $args['length'] ), esc_html( $args['more'] ) );
+}
+
+/**
+ * Echo the copyright text saved in the Customizer.
+ *
+ * @author WDS
+ * @return bool
+ */
+function dctx_display_copyright_text() {
+
+	// Grab our customizer settings.
+	$copyright_text = get_theme_mod( 'dctx_copyright_text' );
+
+	// Stop if there's nothing to display.
+	if ( ! $copyright_text ) {
+		return false;
+	}
+
+	echo dctx_get_the_content( do_shortcode( $copyright_text ) ); // phpcs: xss ok.
```

Comment:
:warning: **Warning**: Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native &quot;phpcs:ignore Standard.Category.SniffName.ErrorCode&quot; annotations instead. Found: // phpcs: xss ok.\n (*WordPress.Security.EscapeOutput.DeprecatedWhitelistCommentFound*).